include/pybind11/eigen.h

```diff
@@ -233,17 +230,22 @@ template <typename props> handle eigen_array_cast(typename props::Type const &sr
                       src.data(), base);
     }
     else {
+        if (base) {
+            // Should be disabled by `cast_impl`.
+            throw cast_error("should not have reached here (dtype=object)");
```

Comment:
Do we need to test this?  We are testing just `dtype=object` in our test which is present in this string?

Reply:
Done. Tried covering this more comprehensively; at this point, I want it to fail mainly; at later points, I would like it to fail at more defined locations. (Added a `TODO`.)